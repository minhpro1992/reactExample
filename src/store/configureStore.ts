import { createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { composeWithDevTools } from 'redux-devtools-extension'
import { createRootReducer, rootsaga } from './'

const sagaMiddleware = createSagaMiddleware()
const composeEnhancers = composeWithDevTools({})
export default function configureStore(initialState: any) {
    const store = createStore(
        createRootReducer(),
        initialState,
        composeEnhancers(applyMiddleware(sagaMiddleware))
    )
    sagaMiddleware.run(rootsaga)
    return store
}