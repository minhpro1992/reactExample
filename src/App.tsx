import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { HomePageComponent } from "./HomePage";
import { LoginComponent } from "./Login";
import { Store } from "redux";
import { ApplicationState } from "./store/configureStore";
interface ProviderProps {
  store: Store<ApplicationState>;
}

function App({ store }: ProviderProps) {
  return (
    <div className="App">
      <Provider store={store}>
        <Switch>
          <Route path="/login" component={LoginComponent}></Route>
          <Route path="/" component={HomePageComponent}></Route>
        </Switch>
      </Provider>
    </div>
  );
}

export default App;
