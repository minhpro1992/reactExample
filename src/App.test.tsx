import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import configureStore, { ApplicationState } from './store/configureStore';

const initialState = window.INITIAL_REDUX_STATE;
const store = configureStore(initialState as ApplicationState);

describe('<App/>', () => {
  test('shoud render App component', async() => {
    render(<App store={store} />)
  })
  test('shoud match snapshot', () => {
    const { asFragment } = render(<App store={store} />);
    expect(asFragment()).toMatchSnapshot();
  })
});
