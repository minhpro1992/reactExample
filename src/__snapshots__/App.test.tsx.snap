// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> render App snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form
      autocomplete="off"
      class="makeStyles-root-1"
      novalidate=""
    >
      <div>
        Login Page
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="username"
        >
          Username
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="currentPassword"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="currentPassword"
            type="currentPassword"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`<App/> shoud match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form
      autocomplete="off"
      class="makeStyles-root-2"
      novalidate=""
    >
      <div>
        Login Page
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="username"
        >
          Username
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="username"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
          for="currentPassword"
        >
          Password
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="currentPassword"
            type="currentPassword"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
